source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -1333,19 +1333,18 @@ OAuth2Filter::validateOAuthCallback(const Http::RequestHeaderMap& headers,
   // More information can be found at https://datatracker.ietf.org/doc/html/rfc6819#section-5.3.5
   std::string csrf_token = filed_value_pair.at(stateParamsCsrfToken).string_value();
   if (!validateCsrfToken(headers, csrf_token)) {
-    ENVOY_LOG(error, "csrf token validation failed");
-    return {false, "", ""};
+    return {false, "", "", "CSRF token validation failed"};
   }
   const std::string original_request_url = filed_value_pair.at(stateParamsUrl).string_value();
 
   // Return 401 unauthorized if the URL in the state is not valid.
   Http::Utility::Url url;
   if (!url.initialize(original_request_url, false)) {
-    ENVOY_LOG(error, "state url {} can not be initialized", original_request_url);
-    return {false, "", ""};
+    return {false, "", "",
+            fmt::format("State url {} can not be initialized", original_request_url)};
```

Comment:
All but two details args place there parameters after the first `: ` if there are paramters. May I suggest that this is changed to `State url can not be initialized: {}`. That will make counting easier.